pdm/models/markers.py

```diff
@@ -120,7 +121,13 @@ def split_marker_extras(
     return result, marker
 
 
-def _only_contains_python_keys(markers):
+def _only_contains_python_keys(
+    markers: Union[
+        List[Tuple[Variable, Op, Value]],
+        List[Union[Tuple[Variable, Op, Value], str]],
+        Tuple[Variable, Op, Value],
```

Comment:
For such type annotations which is too verbose, I would like to remove them.

Reply:
I get it.
I'll look the codes again and reduce too verbose annotations.

Reply:
fixed here

[`312287c` (#354)](https://github.com/pdm-project/pdm/pull/354/commits/312287c375db739caa180500afbc976002515113#diff-79c8eb95336c3a619cac203017451bbd32680ac013513a013b2c38087fc6ed87R123)

I checked how markers should be typed here.
https://gitlab.u-angers.fr/projetsetudiants/3a_2021_trafic2/-/blob/f593f8e345fec6a206816b8212675af186e06c9f/Traffik0.1/Lib/site-packages/pip/_vendor/packaging/markers.py#L299